### PR TITLE
Adding ETH and multiple ERC20 payment support

### PR DESCRIPTION
**Pull request by aaitor**
## Description

Adding ETH and multiple ERC20 support for payment in Lock and Reward payment conditions

## Is this PR related with an open issue?

Related to Issue #

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)
 
## Checklist:

- [X] Follows the code style of this project.
- [X] Tests Cover Changes
- [X] Documentation

### CODE REVIEW

File: contracts/conditions/rewards/EscrowPaymentCondition.sol [hunk 1]
```diff
@@ -231,4 +240,37 @@ contract EscrowPaymentCondition is Reward {
...
+    {
+
+        for(uint i = 0; i < _receivers.length; i++)    {
+            require(
+                _receivers[i] != address(this),
+                'Escrow contract can not be a receiver'
+            );
+            (bool sent,) = _receivers[i].call{value: _amounts[i]}("");
+            require(sent, "Failed to send Ether");
+//            _receivers[i].transfer(_amounts[i]);
```
Comment (mrsmkl) [inline contracts/conditions/rewards/EscrowPaymentCondition.sol:267]: Probably best to decide which one to use (for transferring ETH), and then use it here and `LockPaymentCondition` too.

File: contracts/conditions/LockPaymentCondition.sol [hunk 2]
```diff
@@ -141,9 +148,36 @@ contract LockPaymentCondition is Condition {
...
+        return true;
+    }
+    
+    function _transferETH(
+        address payable _rewardAddress,
+        uint256 _amount
+    )
+    internal
+    {
+        _rewardAddress.transfer(_amount);
```
[the remark this answers is not in the document]
Comment (aaitor) [reply contracts/conditions/LockPaymentCondition.sol:179]: ya, `LockPaymentCondition` doesn't have funds and when the user execute the fulfill it locks the funds in the `_rewardAddress`, typically the `EscrowPaymentCondition.address`

File: contracts/conditions/rewards/EscrowPaymentCondition.sol (hunk 1 above)
Comment (aaitor) [reply contracts/conditions/rewards/EscrowPaymentCondition.sol:267]: The payment flow support distribute multiple rewards (`_amounts`) to multiple addresses (`receivers`). So the fulfill condition do the transfers using the previously locked funds by the user in the contract using the `LockPaymentCondition`.

Comment (aaitor) [reply contracts/conditions/rewards/EscrowPaymentCondition.sol:267]: Full flow can be found here: https://github.com/nevermined-io/contracts/blob/feature/eth_payment/test/unit/conditions/rewards/EscrowPayment.Test.js#L204

File: contracts/conditions/rewards/EscrowReward.sol [hunk 3]
```diff
@@ -21,6 +21,7 @@ import '../ConditionStoreLibrary.sol';
  */
 contract EscrowReward is Reward {
```
Comment (r-marques) [inline contracts/conditions/rewards/EscrowReward.sol:22]: Any reason to still keep the EscrowReward?

**Review by r-marques — APPROVED** [review]
lgtm

File: contracts/conditions/rewards/EscrowReward.sol (hunk 3 above)
Comment (aaitor) [reply contracts/conditions/rewards/EscrowReward.sol:22]: Good point, I agree we should remove some stuff. I think I will open an issue to deprecate this and some other stuff as part of a different PR.

**Pull request merged**